src/commands/protocols/common/epoch-command.js

```diff
@@ -17,18 +17,20 @@ class EpochCommand extends Command {
         agreementData,
         operationId,
     ) {
-        // todo check epoch number and make sure that delay is not in past
-        const nextEpochStartTime =
-            Number(agreementData.startTime) + Number(agreementData.epochLength) * (epoch + 1);
-
-        // delay by 10% of commit window length
-        const offset =
+        let newEpochNumber = epoch;
+        let delay = -1;
+        const commitWindowDurationOffset =
             Number(await this.blockchainModuleManager.getCommitWindowDuration(blockchain)) * 0.1;
-
-        const delay = nextEpochStartTime - Math.floor(Date.now() / 1000) + offset;
+        while (delay < 0) {
```

Comment:
can't we avoid the while loop by doing (time now - agreement start time) / epoch length ?